Add Lottie

Soare-Robert-Daniel opened the pull request:
# Lottie Animations
> Lottie is a library for Android, iOS, Web, and Windows that parses Adobe After Effects animations exported as JSON with Bodymovin and renders them natively on mobile and on the web!

### Technical Details

##### Libraries Used
- [Lottie React from LottieFiles](https://github.com/LottieFiles/lottie-react)
- [Lottie Player from LottieFiles](https://github.com/LottieFiles/lottie-player)

##### Lottie Animation Vendors
- [LottieFiles](https://lottiefiles.com/)

### Available Settings
- Autoplay
- Loop
- Custom Background
- Animation Speed
- Custom Size
- Show Controls Option
- Renderer Option

### Known Issues
- [ ] When the size is too big it tends to overflow others components when is used in a Section Blocks
- [ ] It doesn't work with AMP

### To Do
- [ ] New error systems when the source is invalid
    - [ ] Improve visual effects for displaying the error
    - [ ] Improve error detection system
- [ ] New design
    - [x] Improve the design of the Block
    - [ ] Improve the design of the Inspector
- [ ] Additional method for adding animations

Soare-Robert-Daniel committed:
Add Lottie

HardeepAsrani commented:
Initial review: https://www.loom.com/share/bcf35fcc29bc40b1a8da584b0da97535

HardeepAsrani commented:
Apart from this, you add it to `themeisle-blocks` category which won't be available always, if someone is not using Otter but only using our Animations block. So check if it exists if not then create one.

Soare-Robert-Daniel committed:
Add new Lottie version and config

Soare-Robert-Daniel committed:
Minor Improvments

Soare-Robert-Daniel committed:
Add option to load animation from Media Library

Soare-Robert-Daniel committed:
Add direct insertion

Soare-Robert-Daniel committed:
Load lottie-player.js and build/style.css

HardeepAsrani committed:
Improve build

HardeepAsrani committed:
Remove unused package

HardeepAsrani committed:
Remove unused package

HardeepAsrani committed:
Remove background color

HardeepAsrani commented:
@Soare-Robert-Daniel I've removed background color option as it doesn't do anything special but to add a background to the container, which can be done with a Group or Section block.

HardeepAsrani commented:
Also, the mouse hover & looping doesn't work in the editor when you change those options.

HardeepAsrani committed:
Restructure files

HardeepAsrani committed:
Add ID mechanism

HardeepAsrani committed:
Improve Placeholder

Instead of using Placeholder component, we use MediaPlaceholder to
upload and manage files.

HardeepAsrani commented:
In this https://github.com/Codeinwp/gutenberg-animation/pull/137/commits/a431e599a3729c3ddeaeac05aafe076788041f0f, I simplified the code to improve UI and flow of things. It looks like this now:

<img width="700" alt="Screen Shot 2020-08-13 at 3 41 48 AM" src="https://user-images.githubusercontent.com/2649903/90073761-6a8f8b80-dd17-11ea-8223-e905d639162a.png">

Soare-Robert-Daniel committed:
Minor fix on loop

HardeepAsrani committed:
Change attribute to object

HardeepAsrani commented:
I did some try to see why Hover wasn't working and it seems Lottie doesn't include those props. For that, we will have to include https://github.com/LottieFiles/lottie-interactivity to our project.
